### PR TITLE
Update/py3 13 deps

### PR DESCRIPTION
**Pull request by gmorales96**
### Description

This PR updates the testing setup and ensures compatibility with supported Python versions.

### Key Changes

1. **Updated Testing Dependencies**: All testing-related dependencies have been updated to their latest compatible versions.
2. **Added Python Version Support**: Testing now includes Python versions 3.9, 3.10, 3.11, 3.12, and 3.13.
3. **Dropped Python 3.8 Support**: Consistent with other projects, support for Python 3.8 has been removed.

### Motivation

These changes align with the project's goal of maintaining consistency across supported versions and ensuring up-to-date dependency management. Dropping Python 3.8 reflects the end of its upstream support and reduces maintenance overhead.

## Release Notes v0.2.3.dev4

- **New Features**
  - Added support for Python 3.9-3.13
  - Expanded test coverage for edge cases, including a new test for handling `None` input in helper functions

- **Dependencies**
  - Updated development dependencies to latest versions, including `black`, `flake8`, `isort`, `mypy`, `pytest`, and others
  - Refined version constraints for `mongoengine`, `dnspython`, `pymongo`, and `pymongocrypt`
  - Updated `boto3` version constraints

- **Testing**
  - Enhanced test matrix to include multiple Python versions
  - Enhanced type-checking for `ssn` field in tests

- **Maintenance**
  - Minor formatting improvements in JSON output
  - Updated project configuration files, including Python version requirements and GitHub Actions workflows
  - Incremented version number to reflect the new stable release

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
       - name: Set up Python
         uses: actions/setup-python@v2.2.2
         with:
-          python-version: 3.8
+          python-version: 3.13
```
[the remark this answers is not in the document]
Comment (pachCode) [reply .github/workflows/test.yml:13]: es buen comentario @gmorales96

File: tests/models/test_helpers.py [hunk 2]
```diff
@@ -103,3 +103,8 @@ def test_mongo_to_dict():
         'lastname': 'other',
     }
     assert model_dict['code'] == 0
+
+
+def test_mongo_to_dict_with_none():
+    result = mongo_to_dict(None)
+    assert result == {}
```
Comment (felipao-mx) [inline tests/models/test_helpers.py:110]: este test es de alguna parte del código que no tiene coverage?

**Review by felipao-mx — APPROVED** [review]
en general lo veo bien, solo tengo una duda con un test

**Review by pachCode — APPROVED** [review]
